src/test/scala/memeid/V3Spec.scala

```diff
@@ -0,0 +1,23 @@
+package memeid
+
+import cats.effect._
+import cats.syntax.eq._
+
+import org.specs2.ScalaCheck
+import org.specs2.matcher.IOMatchers
+import org.specs2.mutable.Specification
+
+class V3Spec extends Specification with ScalaCheck with IOMatchers {
+  "V3 constructor" should {
+    "create same UUID for the same namespace/name" in {
```

Comment:
Maybe add a test that checks the opposite (distinct UUIDs for distinct namespace/name)